storage/namespace.go

```diff
@@ -537,8 +582,10 @@ func (n *dbNamespace) NeedsFlush(blockStart time.Time) bool {
 		if shard == nil {
 			continue
 		}
-		if shard.FlushState(blockStart).Status == fileOpInProgress {
-			return false
+		for _, blockStart := range blockStarts {
```

Comment:
This means if any shard is flushing at any time within the range, we are not going to flush any shard for any start times within that range, which seems a bit strange.